src/DictVectors/initiators.jl

```diff
@@ -0,0 +1,219 @@
+    return InitiatorValue{U}(safe=x.safe, unsafe=x.unsafe, initiator=x.initiator)
+end
+function Base.convert(::Type{<:InitiatorValue{V}}, x) where {V}
+    return InitiatorValue{V}(safe=convert(V, x))
+end
+
+"""
+    AbstractInitiator
+
+Abstract type for defining initiator rules. Concrete implementations:
+
+* [`Initiator`](@ref)
+* [`SimpleInitiator`](@ref)
+* [`CoherentInitiator`](@ref)
+"""
+abstract type AbstractInitiator{V} end
+
+"""
+    Initiator
+
+Initiator to be passed to [`InitiatorDVec`](@ref). Initiator rules:
+
+* Initiators can spawn anywhere.
+* Non-initiators can spawn to initiators.
+"""
+struct Initiator{V} <: AbstractInitiator{V}
+    threshold::V
+end
+
+function value(i::Initiator, v::InitiatorValue)
+    return v.safe + v.initiator + !iszero(v.initiator) * v.unsafe
+end
+
+"""
+    SimpleInitiator
+
+Simplified initiator to be passed to [`InitiatorDVec`](@ref). Initiator rules:
+
+* Initiators can spawn anywhere.
+* Non-initiators cannot spawn.
+"""
+struct SimpleInitiator{V} <: AbstractInitiator{V}
+    threshold::V
+end
+
+function value(i::SimpleInitiator, v::InitiatorValue)
+    return v.safe + v.initiator
+end
+
+"""
+    CoherentInitiator
+
+Initiator to be passed to [`InitiatorDVec`](@ref). Initiator rules:
+
+* Initiators can spawn anywhere.
+* Non-initiators can spawn to initiators.
+* Multiple non-initiators can spawn to a single non-initiator if their contributions add up
+  to a value greater than the initiator threshold.
+"""
+struct CoherentInitiator{V} <: AbstractInitiator{V}
+    threshold::V
+end
+
+function value(i::CoherentInitiator, v::InitiatorValue)
+    if !iszero(v.initiator) || abs(v.unsafe) > i.threshold
+        return v.initiator + v.safe + v.unsafe
+    else
+        return v.initiator + v.safe
+    end
+end
+
+"""
+    InitiatorDVec
+
+Functionally identical to [`DVec`](@ref), but contains [`InitiatorValue`]s
+"""
+struct InitiatorDVec{
+    K,V,D<:Dict{K,InitiatorValue{V}},S<:StochasticStyle{V},I<:AbstractInitiator
+} <: AbstractDVec{K,V}
+    storage::D
+    style::S
+    initiator::I
+end
+
+###
+### Constructors
+###
+function InitiatorDVec(
+    dict::Dict{K,V}; style=default_style(V), initiator=Initiator(one(V)), capacity=0
+) where {K,V}
+    T = eltype(style)
+    storage = Dict{K,InitiatorValue{T}}()
+    sizehint!(storage, capacity)
+    for (k, v) in pairs(dict)
+        storage[k] = InitiatorValue{T}(safe=v)
+    end
+    return InitiatorDVec(storage, style, initiator)
+end
+function InitiatorDVec(args...; kwargs...)
+    storage = Dict(args...)
+    return InitiatorDVec(storage; kwargs...)
+end
+function InitiatorDVec(
+    dv::AbstractDVec{K,V}; style=StochasticStyle(dv), initiator=Initiator(one(V)), capacity=0
+) where {K,V}
+    storage = Dict{K,InitiatorValue{V}}()
+    sizehint!(storage, max(capacity, length(dv)))
+    for (k, v) in pairs(dv)
+        storage[k] = InitiatorValue{V}(safe=v)
+    end
+    return InitiatorDVec(storage, style, initiator)
+end
```

Comment:
So, if we call this constructor on an `InitiatorDVec`, then we will get a copy where all the information within the `InitiatorValue`s is lost. Is that what we want? Would it not make more sense to just get the same object back?